app/assets/javascripts/tag_by_tag_plate.js

```diff
@@ -17,21 +17,29 @@
 
     qcLookup = function(barcodeBox, collector) {
       if (barcodeBox.length === 0) { return false; }
+
       var qc_lookup = this, status;
       this.inputBox = barcodeBox;
+
+      // the `data` attribute is set when declaring the element in tagged_plate,html.erb
       this.infoPanel = $('#'+barcodeBox.data('info-panel'));
       this.dualIndex = barcodeBox.data('dual-index');
       this.approvedTypes = SCAPE[barcodeBox.data('approved-list')];
       this.required = this.inputBox[0].required;
+
+      // add an onchange event to the box, to look up the plate
       this.inputBox.on('change', function(){
         qc_lookup.resetStatus();
         qc_lookup.requestPlate(this.value);
       });
       this.monitor = collector.register(!this.required, this);
+
+      // set initial values for the tag plate data
       this.qcable = unkownQcable;
       this.template = unknownTemplate;
     };
 
+    // why is this called `qcLookup`? Shouldn't it be tagPlateLookup or similar?
     qcLookup.prototype = {
```

Comment:
Because it fetches the 'qcable' rather than the plate itself, this could be a tag tube, tag plate (or potentially also a reporter, although that wouldn't get approved).

Reply:
OK, and why is a qcable called a qcable? Is a tag plate something to do with quality control? What's a reporter? :)

Reply:
(Reply from Slack):
Yep, so tag plates were introduced along with Gatekeeper, which was developed to add a QC process for tags. We had large batches of tags coming in from a supplier, which were then stamped out into individual tag plates, which underwent an internal QC process in Gatekeeper. These were the QCables, which would then either be released for use in labs, or failed, if they showed cross contamination. Reporters are short sequences of DNA that are used in this QC process, and themselves needed to be QCed.

Gatekeeper was so named because it was literally the gatekeeper of the tags that could go into Sequencescape. ANd QCable was anything that passed through its QC process.

Only… well now IDT mostly QC their own tags, and I’m not sure the QC component of Gatekeeper is even used nay-more

And until very recently we weren’t even stamping tags out any-more… and I don’t think the current stamping process actually uses the stamping pipleine.

Instead all plates come in ready stamped, and ready passed. Making the name Gatekeeper a bit irrelevant, and the QC name a bit confusing.